src/Frontend/Testing.hs

```diff
@@ -0,0 +1,19 @@
+{-# OPTIONS_GHC -Werror -Wall #-}
+
+module Frontend.Testing
+where
+
+import Control.Lens ((^.))
+
+import Action
+import Persistent
+import Types
+
+-- | Make a topic timeout if the timeout is applicable.
+makeTopicTimeout :: (ActionPersist r m, ActionUserHandler m) => AUID Topic -> m ()
+makeTopicTimeout tid = do
+    Just topic <- persistent $ findTopic tid -- FIXME: 404
```

Comment:
ah, some previous comments do not apply any more.  but the FIXME can be removed: it's testing code, and it is not supposed to be fixed here anyway.